### PR TITLE
Fix authenticator nil pointer error

### LINKED ISSUE #53: Authenticator nil pointer cause eve-agent panic error
We need to avoid panic error which will terminate the container and leave our resource in limbo state.

Investigate what caused authenticate nil reference error
https://github.com/concur/eve/blob/master/provider/aws/authenticator.go#L33
```
{"log":"panic: runtime error: invalid memory address or nil pointer dereference\n","stream":"stderr","time":"2017-06-30T19:54:21.740650051Z"}
{"log":"[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x824089]\n","stream":"stderr","time":"2017-06-30T19:54:21.740665705Z"}
{"log":"\n","stream":"stderr","time":"2017-06-30T19:54:21.740671277Z"}
{"log":"goroutine 20 [running]:\n","stream":"stderr","time":"2017-06-30T19:54:21.740675745Z"}
{"log":"github.com/concur/eve/provider/aws.(*Authenticator).Authenticate(0xc420287cb0, 0xc420376f70, 0x0, 0x0)\n","stream":"stderr","time":"2017-06-30T19:54:21.740680446Z"}
{"log":"\u0009/go/src/github.com/concur/eve/provider/aws/authenticator.go:33 +0x29\n","stream":"stderr","time":"2017-06-30T19:54:21.740685337Z"}
{"log":"github.com/concur/eve/pkg/terraform.(*Terraform).addProviderCredEnv(0xc4204119a8, 0x0, 0x0, 0x0, 0x2, 0xc42022c2c0, 0x1, 0x2, 0x0)\n","stream":"stderr","time":"2017-06-30T19:54:21.740690278Z"}
{"log":"\u0009/go/src/github.com/concur/eve/pkg/terraform/terraform.go:394 +0x47\n","stream":"stderr","time":"2017-06-30T19:54:21.740694812Z"}
{"log":"github.com/concur/eve/pkg/terraform.(*Terraform).runTerraform(0xc4204119a8, 0xa1ec94, 0x5, 0xc4204dbbd0, 0x2f)\n","stream":"stderr","time":"2017-06-30T19:54:21.740708163Z"}
{"log":"\u0009/go/src/github.com/concur/eve/pkg/terraform/terraform.go:358 +0x328\n","stream":"stderr","time":"2017-06-30T19:54:21.740713004Z"}
{"log":"github.com/concur/eve/pkg/terraform.(*Terraform).ApplyQuoin(0xc4204119a8, 0x0, 0x0)\n","stream":"stderr","time":"2017-06-30T19:54:21.740717448Z"}
{"log":"\u0009/go/src/github.com/concur/eve/pkg/terraform/terraform.go:118 +0x310\n","stream":"stderr","time":"2017-06-30T19:54:21.740721737Z"}
{"log":"github.com/concur/eve/cmd/eve/command/agent.create.func2(0xc42023a000)\n","stream":"stderr","time":"2017-06-30T19:54:21.74072597Z"}
{"log":"\u0009/go/src/github.com/concur/eve/cmd/eve/command/agent/create.go:68 +0x5fc\n","stream":"stderr","time":"2017-06-30T19:54:21.74073029Z"}
{"log":"reflect.Value.call(0x977b00, 0xc4201288d0, 0x13, 0xa1e4ba, 0x4, 0xc420375ec8, 0x1, 0x1, 0xc42023a001, 0x9a5e60, ...)\n","stream":"stderr","time":"2017-06-30T19:54:21.740734728Z"}
{"log":"\u0009/usr/local/go/src/reflect/value.go:434 +0x91f\n","stream":"stderr","time":"2017-06-30T19:54:21.740739561Z"}
{"log":"reflect.Value.Call(0x977b00, 0xc4201288d0, 0x13, 0xc420375ec8, 0x1, 0x1, 0x9a5e60, 0xc42023a000, 0x0)\n","stream":"stderr","time":"2017-06-30T19:54:21.740743817Z"}
{"log":"\u0009/usr/local/go/src/reflect/value.go:302 +0xa4\n","stream":"stderr","time":"2017-06-30T19:54:21.740748174Z"}
{"log":"github.com/concur/eve/vendor/github.com/nats-io/go-nats.(*EncodedConn).subscribe.func1(0xc4204c6050)\n","stream":"stderr","time":"2017-06-30T19:54:21.740752394Z"}
{"log":"\u0009/go/src/github.com/concur/eve/vendor/github.com/nats-io/go-nats/enc.go:223 +0x12a\n","stream":"stderr","time":"2017-06-30T19:54:21.740756718Z"}
{"log":"github.com/concur/eve/vendor/github.com/nats-io/go-nats.(*Conn).waitForMsgs(0xc420111680, 0xc4200ba180)\n","stream":"stderr","time":"2017-06-30T19:54:21.740761031Z"}
{"log":"\u0009/go/src/github.com/concur/eve/vendor/github.com/nats-io/go-nats/nats.go:1449 +0x17b\n","stream":"stderr","time":"2017-06-30T19:54:21.740765335Z"}
{"log":"created by github.com/concur/eve/vendor/github.com/nats-io/go-nats.(*Conn).subscribe\n","stream":"stderr","time":"2017-06-30T19:54:21.740769631Z"}
{"log":"\u0009/go/src/github.com/concur/eve/vendor/github.com/nats-io/go-nats/nats.go:1958 +0x4cc\n","stream":"stderr","time":"2017-06-30T19:54:21.740774012Z"}
```

### PR DESCRIPTION
**Pull request by weitenghuang**
**What this PR does**:
This PR is to fix #53 
Due to 
1. Incorrect provider data seeding, **_or_**
2. Incorrect user input _provider slug_ value

[`GetAccount(name string)`](https://github.com/scipian/eve/blob/342aa93e85e2f73eca08f1df4eef00060b02b289/provider/aws/provider.go#L46) might return `<nil>` pointer for `*Account`, which caused the nil pointer panic error if we didn't check `nil` before we assess `a.Account.AuthType` in [`Authenticate()`](https://github.com/scipian/eve/blob/342aa93e85e2f73eca08f1df4eef00060b02b289/provider/aws/authenticator.go#L33)

This PR will add nil check before assess `Account.AuthType`

**Which issue this PR fixes**:
Fixes #53 

**Release note**:
  - `eve/provider/aws` package `*Authenticator Authenticate()` method should have
nil check for Account property
  - Add unit tests for Authenticate method for nil Account, empty Account, unspported AuthType

### CODE REVIEW

File: provider/aws/authenticator_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		authn := NewAuthenticator(acct, "testIamRole", "testRoleSessionName")
+		if cred, err := authn.Authenticate(); err != nil {
+			t.Error("Authenticator with empty account will get credentials from iamDirect()")
+		} else if cred.AccessKeyId != DUMMYKEY || cred.SecretAccessKey != DUMMYACCESS {
+			t.Error("Authenticator got mismatched credential value")
+		}
+	}
+
+	var with_Unsupported_Type = func(t *testing.T) {
+		acct := &Account{AuthType: 99999}
```
Comment (weitenghuang) [inline provider/aws/authenticator_test.go]: I made assumption that we will not have more than **_99998_** iota value [`AuthType`](https://github.com/scipian/eve/blob/342aa93e85e2f73eca08f1df4eef00060b02b289/provider/aws/account.go#L8). I think it might be better off if we use iota=0 as preserved **default value** for future design.
